Ghidra/Framework/Utility/src/main/java/ghidra/GhidraApplicationLayout.java

```diff
@@ -191,7 +191,9 @@ protected ResourceFile findGhidraApplicationInstallationDir() {
 			ResourceFile classpathEntryParent = classpathEntry.getParentFile();
 			if (classpathEntryParent != null &&
 				ModuleUtilities.isModuleDirectory(classpathEntryParent)) {
-				moduleRootDirectories.add(classpathEntryParent);
```

Comment:
> Minor nitpick: I cannot see why we would want duplicates in this 'moduleRootDirectories' collection at all. If that is the case, then line 148 should be using a Set instead. It is not clear if order is important, assuming it is, line 148 can be changed to use a `LinkedHashSet`.
> 
> Will you change your code to my suggestion and see if that works for you?

I initially considered doing so but it would only make sense to do so for the directories. `ModuleUtilities.findModules` checks for duplicates and throws an AssertException if one is found. This makes sense to do because if there is a duplicate module or extension installed the one that is expected to be loaded may not be the one which is actually loaded. The change here is really only to bypass an issue that would arise during development/debugging. This code is also only reached when the entry is a directory and not a jar so it isn't reached very often.

I could create a `LinkedHashSet` before the for loop, fill it with the directory entries, and then utilize `Collection.addAll` to add the sets entries to moduleRootDirectories.

Reply:
I see.  I will have to look more closely at this next week.

Reply:
> I initially considered doing so but it would only make sense to do so for the directories. ModuleUtilities.findModules checks for duplicates and throws an AssertException if one is found. 

So I looked into this.  The intent of the code you mentioned is to handle two modules that share the same name, but live in different paths.    The validation code should work as expected if we switch to a LinkedHashSet implemenation.  If you do that, then you should no longer need the original check that you added.

I can make the change if you are not comfortable doing so.